marketplace/commercetools-products/src/field.js

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Button, Spinner } from '@contentful/forma-36-react-components';
+import { ApolloProvider } from '@apollo/react-components';
+import { SortableContainer } from 'react-sortable-hoc';
+import arrayMove from 'array-move';
+
+import { getApolloClient } from './apollo.js';
+import { SortableProductCard } from './product-card.js';
+
+const SortableList = SortableContainer(function SortableList({
+  items,
+  removeItem,
+  locale,
+  sortable,
+  isDisabled,
+  projectKey
+}) {
+  return (
+    <ol className="product-list">
+      {items.map((sku, index) => {
+        return (
+          <li key={index} className="product-list__item">
+            <SortableProductCard
+              sku={sku}
+              index={index}
+              itemIndex={index}
+              removeItem={removeItem}
+              locale={locale}
+              sortable={sortable}
+              disabled={isDisabled}
+              isDisabled={isDisabled}
+              projectKey={projectKey}
+            />
+          </li>
+        );
+      })}
+    </ol>
+  );
+});
+
+export class CommerceToolsField extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { extension } = props;
+
+    let value = extension.field.getValue();
+    if (value === undefined) {
+      value = [];
+    }
+
+    this.state = {
+      fieldValue: Array.isArray(value) ? value : [value],
+      apolloClient: null,
+      isDisabled: false
+    };
+
+    this.onDialogOpen = this.onDialogOpen.bind(this);
+    this.updateFieldValue = this.updateFieldValue.bind(this);
+    this.onRemoveItem = this.onRemoveItem.bind(this);
+    this.onSortEnd = this.onSortEnd.bind(this);
+  }
+
+  componentDidMount() {
+    const { extension } = this.props;
+
+    getApolloClient(this.props.parameters)
+      .then(apolloClient => {
+        this.setState({ apolloClient });
+      })
+      .catch(err => console.error(err));
+
+    this.unsubscribeOnValue = extension.field.onValueChanged(value => {
+      if (value === undefined) {
+        value = [];
+      }
+
+      this.setState({ fieldValue: Array.isArray(value) ? value : [value] });
+    });
+
+    // Disable editing (e.g. when field is not editable due to R&P).
+    this.unsubscribeOnIsDisabled = extension.field.onIsDisabledChanged(isDisabled => {
+      this.setState({ isDisabled });
+    });
+  }
+
+  componentWillUnmount() {
+    if (this.unsubscribeOnValue) {
+      this.unsubscribeOnValue();
+    }
+
+    if (this.unsubscribeOnIsDisabled) {
+      this.unsubscribeOnIsDisabled();
+    }
+  }
+
+  onDialogOpen() {
+    const { isSingle, extension } = this.props;
+
+    extension.dialogs
+      .openExtension({
+        id: 'commercetools-products', // @todo Check with the team how to make this work with Marketplace installs
```

Comment:
Stable ID is not guaranteed for apps. Use `sdk.ids.extension` to get in on runtime.

Reply:
Works like a charm 👍